pkg/api/docs/merge/merge_specs.go

```diff
@@ -98,6 +135,61 @@ func mergeSpecs(output string, sources ...string) error {
 	return nil
 }
 
+// getDashboardDefinitions converts the dashboard OpenAPI v3 to OpenAPI v2
+// and it returns it as spec.Schema
+func getDashboardDefinitions() (spec.Definitions, error) {
+	lib := thema.NewLibrary(cuecontext.New())
+	lin, _ := dashboard.Lineage(lib)
+	// Grab the 0.0 version. Or whichever one you want
+	sch := thema.SchemaP(lin, thema.SV(0, 0))
+
+	f, _ := openapi.GenerateSchema(sch, nil)
+
+	v3, err := json.Marshal(lib.Context().BuildFile(f))
+
+	loader := openapi3.NewLoader()
+	doc, err := loader.LoadFromData(v3)
+	if err != nil {
+		return spec.Definitions{}, fmt.Errorf("failed to load the OpenAPI v3 dashboard definition, %w", err)
+	}
+
+	if err = doc.Validate(loader.Context); err != nil {
+		return spec.Definitions{}, fmt.Errorf("failed to validate OpenAPI v3 dashboard definition, %w", err)
+	}
+
+	v2, err := openapi2conv.FromV3(doc)
+	if err != nil {
+		return spec.Definitions{}, fmt.Errorf("failed to convert to OpenAPI v2 dashboard definition, %w", err)
+	}
+
+	blob, err := v2.MarshalJSON()
+	if err != nil {
+		return spec.Definitions{}, fmt.Errorf("failed to marshal the OpenAPI v2 dashboard definition, %w", err)
+	}
+
+	// it seems there is a bug in the conversion and there are some unresolved references to #/components/schemas/
```

Comment:
that's frustrating :(

Reply:
file a bug upstream, maybe?

Reply:
The unresolved references are the following so I think that the actual problem is the `oneOf` property that does not exist in v2.
```
					"panels": {
						"items": {
							"oneOf": [
								{
									"$ref": "#/components/schemas/dashboard.Panel"
								},
								{
									"$ref": "#/components/schemas/dashboard.RowPanel"
								},
								{
									"$ref": "#/components/schemas/dashboard.GraphPanel"
								},
								{
									"$ref": "#/components/schemas/dashboard.HeatmapPanel"
								}
							],
							"type": "object"
						},
						"type": "array"
					},
```

In fact, I suspect that's the reason that the `Panels` property looks like this in the generated dashboard model from the v2 spec
<details>
<summary>Generated Dashboard model from the v2 spec</summary>

```
type DashboardModel struct {
	Annotations *DashboardAnnotationsModel `json:"annotations,omitempty"`
	// Description of dashboard.
	Description string `json:"description,omitempty"`
	// Whether a dashboard is editable or not.
	Editable bool `json:"editable"`
	// TODO docs
	FiscalYearStartMonth int32  `json:"fiscalYearStartMonth,omitempty"`
	GnetId               string `json:"gnetId,omitempty"`
	GraphTooltip         int32  `json:"graphTooltip"`
	// Unique numeric identifier for the dashboard. TODO must isolate or remove identifiers local to a Grafana instance...?
	Id int64 `json:"id,omitempty"`
	// TODO docs
	Links []DashboardDashboardLinkModel `json:"links,omitempty"`
	// TODO docs
	LiveNow bool          `json:"liveNow,omitempty"`
	Panels  []interface{} `json:"panels,omitempty"`
	// TODO docs
	Refresh interface{} `json:"refresh,omitempty"`
	// Version of the JSON schema, incremented each time a Grafana update brings changes to said schema. TODO this is the existing schema numbering system. It will be replaced by Thema's themaVersion
	SchemaVersion int32 `json:"schemaVersion"`
	// Theme of dashboard.
	Style string `json:"style"`
	// Tags associated with dashboard.
	Tags       []string                  `json:"tags,omitempty"`
	Templating *DashboardTemplatingModel `json:"templating,omitempty"`
	Time       *DashboardTimeModel       `json:"time,omitempty"`
	Timepicker *DashboardTimepickerModel `json:"timepicker,omitempty"`
	// Timezone of dashboard,
	Timezone string `json:"timezone,omitempty"`
	// Title of dashboard.
	Title string `json:"title,omitempty"`
	// Unique dashboard identifier that can be generated by anyone. string (8-40)
	Uid string `json:"uid,omitempty"`
	// Version of the dashboard, incremented each time the dashboard is updated.
	Version int32 `json:"version,omitempty"`
	// TODO docs
	WeekStart string `json:"weekStart,omitempty"`
}
```

</details>

Reply:
There's no `oneOf` in v2? Oi, that's a challenge. Seems like a bug if 

However, outside of generating a generic Go type (which none of these libs have attempted yet AFAIK), that limitation is going to exist no matter what when trying to generate Go types. `interface{}` is the best that can be done.

Also, we're not trying to generate anything from v2, right? Thema directly produces openapi v3.0 (and implicitly v3.1 as well, as that's how i made it make JSON Schema output), and codegen currently produces the Go type directly from v3.0.